34.py

```diff
@@ -0,0 +1,31 @@
+class Solution:
+    def findIndex(self, nums: List[int], target: int, is_first: bool) -> int:
+        N = len(nums)
+        left, right = 0, N - 1
+
+        while left <= right:
```

Comment:
I'm worried because you use yet another template this time. I don't think you would be able to recall or identify which is which during a contest or an interview. 

In line 6, why do you use `<=` this time?